include/swift/AST/DiagnosticsSema.def

```diff
@@ -992,6 +992,10 @@ ERROR(missing_address_of,none,
 ERROR(missing_address_of_yield,none,
       "yielding mutable value of type %0 requires explicit '&'",
       (Type))
+ERROR(extraneous_property_wrapper_unwrap,none,
+      "property %0 will be unwrapped to value of type %1, use '$' to refer to "
+      "the %2 property wrapper",
```

Comment:
I'm not sure whether "the" is really needed here, I haven't seen it used anywhere else in similar situations. Maybe this should be simplified down to "to refer to wrapper" since message already mentions that this is about the "property" anyway?

Reply:
I think mentioning the type of the property wrapper is reasonably important, especially from a discoverability standpoint. By doing so, it not only points users towards the way to fix the problem, but also makes it clear that using a dollar sign gives you access to an actual instantiation of the wrapping type so users can figure out when they need the `$` going forward.

The exact phrasing, I feel less strongly about. I mostly care that we work the type of the wrapper into the diagnostic somewhere. WDYT?

Reply:
As of now, I changed it to: "property 'x' will be unwrapped to value of type 'Int', use '$' to refer to 'Foo<Int>' wrapper" as a less wordy version that still sneaks in the wrapper type. How does that sound?

Reply:
Maybe `to refer to wrapper type 'Foo<Int>'`?

Reply:
After  thinking about this some more I think in relation to access of properties/methods on wrapper type the best diagnostic would be something like `referencing instance method 'foo(_:)' requires a wrapper type 'Foo<Int>'` and the we don't even have to mention `$` can just add it as a fix-it. But this doesn't have to happen right away, we can do it in a follow-up.

Reply:
Alright, so for now, I'll make it as `to refer to wrapper type Foo<Int>`, and we'll change that message to something more along those lines in a follow up.

Reply:
Yeah, sounds good! We still need a reverse version when $ is extraneous, so we can do both diagnostics then.